src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -65,6 +65,23 @@ private protected override SimpleResolver CreateResolver(PowerFxConfig alternate
         /// <inheritdoc/>
         protected override IExpression CreateEvaluator(CheckResult result)
         {
+            return CreateEvaluatorDirect(result);
+        }
+
+        /// <summary>
+        /// Create an evaluator over the existing binding. 
+        /// </summary>
+        /// <param name="result">A successful binding from a previous call to <see cref="Engine.Check(string, RecordType, ParserOptions)"/>. </param>
+        /// <returns></returns>
+        public static IExpression CreateEvaluatorDirect(CheckResult result)
```

Comment:
Why is this public?

Reply:
So that other Engines (notably dataverse) can run the interpreter on *their* IRs. 
The fundamental reason things need to derive from Engine is to override *the resolver*,  not to interpret.

Ie, there's nothing actually inherently coupled between the RecalcEngine and the interpreter.  (ditto for the other parts of the pipeline like parser, binder, etc)